shell/platform/android/android_surface_gl.cc

```diff
@@ -39,20 +39,27 @@ void AndroidSurfaceGL::TeardownOnScreenContext() {
   // When the onscreen surface is destroyed, the context and the surface
   // instance should be deleted. Issue:
   // https://github.com/flutter/flutter/issues/64414
-  android_context_.ClearCurrent();
+  GLContextPtr()->ClearCurrent();
   onscreen_surface_ = nullptr;
 }
 
 bool AndroidSurfaceGL::IsValid() const {
-  return offscreen_surface_ && android_context_.IsValid();
+  return offscreen_surface_ && GLContextPtr()->IsValid();
 }
 
 std::unique_ptr<Surface> AndroidSurfaceGL::CreateGPUSurface(
     GrDirectContext* gr_context) {
   if (gr_context) {
     return std::make_unique<GPUSurfaceGL>(sk_ref_sp(gr_context), this, true);
+  } else {
+    sk_sp<GrDirectContext> main_skia_context =
```

Comment:
This is more or less 1 of 2 of the actual interesting parts of this PR. When AndroidSurfaceGL needs to make a GPUSurfaceGL (which would then make a GrDirectContext implicitly), the AndroidSurfaceGL makes one explicitly, then stores it in the AndroidContext for reuse if more AndroidSurfaceGLs are made with the same AndroidContext.